asdf/schema.py

```diff
@@ -385,7 +391,20 @@ def load_schema(url, resolver=None, resolve_references=False,
         control local reference resolution separately.
     """
     if resolver is None:
-        resolver = default_ext_resolver
+        # We can't just set this as the default in load_schema's definition
+        # because invoking get_default_resolver at import time leads to a circular import.
+        resolver = extension.get_default_resolver()
+
+    # We want to cache the work that went into constructing the schema, but returning
```

Comment:
Do we have examples of where a schema is changed on the fly to worry about this? If not, is there a good use case you have in mind?

Reply:
One of the tests in this repo does, which leads to thrilling heisenbugs in other tests.  I'm not aware of a real use case but this kind of problem is so difficult to track down that it seemed best to be defensive here.

Reply:
If it were possible to make the schema immutable, that would be my solution (I haven't looked into that). But this will do for now.